blocks/theming/theming.js

```diff
@@ -155,16 +161,28 @@ export default class Theming extends ComponentBase {
   }
 
   async loadFragment() {
-    Promise.all(
-      ['color', 'font', 'layout', 'component'].map(async (fragment) => {
-        const metaKey = `theme${fragment}`;
-
-        const path = getMeta(metaTags[metaKey].metaName) || metaTags[metaKey].fallbackContent;
-        return fetch(`${path}.json`).then((response) => this.processFragment(response, fragment));
+    const themeConfigs = getMetaGroup(metaTags.themeConfig.metaNamePrefix, { getFallback: true });
+
+    // Get the configs async
+    const configsResponses = await Promise.allSettled(
+      themeConfigs.map(async ({ name, content }) => {
+        const response = await fetch(`${content}.json`);
+        if (response.ok) {
+          return {
```

Comment:
Well, I guess the fix was simply adding await before `Promise.all`
But you also want to use the metas configs as getMetaGroup with is a improvement 👍 
here
```javascript
const themeConfigs = getMetaGroup(metaTags.themeConfig.metaNamePrefix, { getFallback: true });
await Promise.all(
      themeConfigs.map(async ({ name, content }) =>
        fetch(`${content}.json`).then((response) => this.processFragment(response, name)),
      ),
    );
this.styles();
```

1 - No need to respond to something, then loop through and skip if there's no type It's just doubling the size of the function with no reason.

2 - Keep processFrament async with response.ok and type there

Reply:
I also though the await should be enough but it seems it doesn't work.
The this.processFragment() must be synchronous because it's using this.variations to save some values which is common between calls and if it's async the calls will overwrite themselves.

Reply:
async will wait, just need to return a value from processFragment
return this.themeJson[type] if the response is ok

Reply:
I also think the defect might be related to an override somewhere instead of a racing condition.
that was not meant to override..
where readValue should extend.
So async would not be the issue in the first place




[2:19](https://ncchat.slack.com/archives/C06E8U7F2AE/p1726229989925899?thread_ts=1726131894.986759&cid=C06E8U7F2AE)
but making Synchronous might resolve if set in the correct order

Reply:
``` JS
await Promise.allSettled(
      themeConfigs.map(async ({ name, content }) =>
        fetch(`${content}.json`).then((response) => this.processFragment(response, name)),
      ),
    );
    
```

The fetch calls in the `.map()` are not waiting for each other to finish they are all async which means `this.processFragment` will/can be called at the same time with other calls processing this.variations  which breaks the theming. 

This was the issue raised, on multiple page reloads the theming will be randomly broken depending on the fetch calls.
 In the initial version there were duplicate calls in a sync way that why it worked and after removing them we had this issue.
![2024-09-13 - Capture - 002409@2x](https://github.com/user-attachments/assets/6265559e-24db-42e1-9de9-3f6469dd8767)

Reply:
async will wait, just need to return a value from processFragment return this.themeJson[type] if the response is 
```javascript
  async processFragment(response, type = 'color') {
    if (response.ok) {
      const responseData = await response.json();
      this.themeJson[type] = responseData;
      if (type === 'fontface') {
        this.fontFaceTemplate(responseData);
      } else if (type === 'component') {
        Object.keys(responseData).forEach((key) => {
          if (key.indexOf(':') === 0 || responseData[key].data.length === 0) return;
          this.componentsConfig[key] = this.componentsConfig[key] || {};
          this.componentsConfig[key] = readValue(responseData[key].data, this.componentsConfig[key]);
        });
      } else {
        this.variations = readValue(responseData.data, this.variations);
        this.defineVariations();
      }
      return this.themeJson[type];
    }
    return false;
  }
  ```